test/Pool.test.js

```diff
@@ -1315,6 +1315,196 @@ describe('Pool', function () {
     });
   });
 
+  it('should not terminate worker if abort listener is defined dedicated worker with Timeout', function () {
+    var workerCount = 0;
+    var pool = createPool(__dirname + '/workers/cleanup-abort.js', {
+      maxWorkers: 1,
+      onCreateWorker: () => {
+        workerCount += 1;
+      }
+    });
+    
+    return pool.exec('asyncTimeout', [],  {
+    })
+    .timeout(200)
+    .catch(function (err) {
+      assert(err instanceof Promise.TimeoutError);
+      let stats = pool.stats();
+      assert(stats.activeTasks === 0);
+    }).then(function() { 
+      return pool.exec(add, [1, 2]) 
+    }).then(function() {
+      assert.strictEqual(workerCount, 1);
+    });
+  });
+
+  it('should not terminate worker if abort listener is defined dedicated worker with Cancellation', function () {
+    var workerCount = 0;
+    var pool = createPool(__dirname + '/workers/cleanup-abort.js', {
+      maxWorkers: 1,
+      onCreateWorker: () => {
+        workerCount += 1;
+      }
+    });
+    
+    let task = pool.exec('asyncTimeout', [],  {});
+
+    // Wrap in a new promise which waits 50ms
+    // in order to allow the function executing in the
+    // worker to 
+    return new Promise(function(resolve) {
+      setTimeout(function() {
+        resolve();
+      }, 50);
```

Comment:
Maybe overkill for this unit test, but instead of a "blind" waiting of 50 ms, you could let the worker emit an event as soon as it started executing the task, and use that as trigger.

see: [Events](https://github.com/josdejong/workerpool?tab=readme-ov-file#events)

Reply:
I tried using an event, however after some digging into the implementation shown [here](https://github.com/josdejong/workerpool/blob/cdb5861133fbce9c911545308871b81ecb6a36cd/src/WorkerHandler.js#L273). I found that is not possible without refactoring to handle `tacked` tasks since the `task` queue is checked and if a task is not found we simply continue processing tasks.

Reply:
Ok let's not put effort in that just for the sake of the unit tests, thanks for checking this out.